doc/adrs/022_add_support_for_selection_dialog.adoc

```diff
@@ -0,0 +1,129 @@
+We will need to indicate that a tool requires a selection dialog instead of relying on the `Composite Processor` label.
+For that, we will add the field `selectionId` to CreateNodeTool:
+
+```
+type CreateNodeTool {
+  id: ID!
+  label: String!
+  imageURL: String!
+  appliesToDiagramRoot: Boolean!
+  targetDescriptions: [NodeDescription!]!
+  selectionId: String
+}
+```
+
+If the id is null, there is nothing to do, otherwise, we will open a selection dialog to display the selection representation with the given id.
+
+
+=== Backend integration
+
+One project, `sirius-web-selection`, will be created for the representation and its description.
+Its content will look like the projects `sirius-web-trees` or `sirius-web-forms`.
+It will contain the data structure of the representation, its description, its renderer, components etc.
+
+The data structure of the selection representation will look like this:
+
+```
+type Selection {
+  id: UUID
+  message: String!
+  objects: [SelectionObject!]!
+}
+
+type SelectionObject {
+  id: String!
+  label: String!
+}
+```
+
+The description of the representation will look like this:
+
+```
+public class SelectionDescription {
+  id: UUID
+  label: String
+  canCreatePredicate: variableManager -> false // To prevent it from being created in the explorer
+  messageProvider: Function<VariableManager, String>
+  objectsProvider: Function<VariableManager, List<Object>>
+  idProvider: Function<VariableManager, String> // Calls IObjectService#getId in our own instance of the description
+  labelProvider: Function<VariableManager, String> // Calls IObjectService#getLabel in our own instance of the description
+}
+```
+
+When we will provide an implementation of this description, in our sample application project, the `idProvider` and `labelProvider` will probably call `IObjectService#getId` and `IObjectService#getLabel` respectively.
+
+Two new projects, `sirius-web-collaborative-selection-api` and `sirius-web-spring-collaborative-selection`, will be created for the subscription.
+They will look like the projects `sirius-web-collaborative-trees-api` and `sirius-web-spring-collaborative-trees` with their configuration, input, payload, event processor factory and event processor.
+
+In a similar fashion as for the diagrams, a type provider will have to be created (like `DiagramTypesProvider`) in order to register our new representation in the GraphQL schema.
+We will need to be careful with the GraphQL annotation on our POJOs (Selection, SelectionObject, SelectionDescription, etc).
+A DataFetcher will also have to be created for the representation, the `SubscriptionSelectionEventDataFetcher` in a similar fashion as the `SubscriptionTreeEventDataFetcher`.
+In order to create the GraphQL subscription, an event processor and its factory will be needed just like the Tree representation for example.
+
+
+=== Execution
+
+When a tool will be executed, the frontend will give back to the backend the id of the selected element.
```

Comment:
... of the selected element of the selection dialog.

Reply:
Added to the document